tracer/test/Datadog.Trace.TestHelpers/FluentAssertionsExtensions/Json/JTokenAssertions.cs

```diff
@@ -44,6 +50,12 @@ public JTokenAssertions(JToken subject)
 
     private GenericCollectionAssertions<JToken> EnumerableSubject { get; }
 
+    [ModuleInitializer]
```

Comment:
How exactly did you track this down? Debug step through when running the tests?

Reply:
> How exactly did you track this down

Looking more closely at the stack trace in the failure in ciapp, it showed that a collection was modified while executing the fluentassertions method `FluentAssertions.Formatting.Formatter.Format()`. From there I looked to see what was being enumerated, and found the mutable collection `Formatter.Formatters`. Did a **Find all usages** on that, and discovered that we _are_ modifying the collection in the json assertion extensions.

I think this would have been tricky to track down as it's very much timing related - we only hit the bug _if_ we run a test that was using `ContainEquivalent` (or some other equivalency) _at the same time_ as the JSON assertions is initializing the collection. If it does the initialization before we start enumerating, we're fine. If the tests don't run at the same time, we're fine. So a bit tricky!